### ISSUE TITLE
Performance Issue

### ISSUE THREAD

**Issue by yenmoc** [hadashiA/VitalRouter#10]
Unity 2022.3.12f1
VitalRouter v0.2.0

I tried checking with profiler, it seems that the `PublishAsync` command is quite slow and it allocates each publish, maybe I set it up wrong?

```csharp
    private DataPresentor _dataPresentor;

    public void Start()
    {
        _dataPresentor = new DataPresentor();
        var subscription = _dataPresentor.MapTo(Router.Default);
        Profiler.BeginSample("Test");
        Router.Default.PublishAsync(new DataCommand() {Damage = 100});
        Profiler.EndSample();
        subscription.Dispose();
    }
```

![image](https://github.com/hadashiA/VitalRouter/assets/44673303/29e83008-5a33-4d25-bd62-781a0020e5da)

**Comment by hadashiA** [comment]
Thanks for the report.

The premise is that allocations occur the first time because of object pooling, caching, or permanent buffer allocation. (This is also true for UniTask's AsyncMethodBuidler)

So, when measuring, please look at the time before and after the second warmed up.


However, upon investigation, we found one problem.

Oddly enough, if I set the type parameter T to `is OtherType x`, the allocation is reported on Unity.
( This does not happen in the dotnet environment, I think )

I will make an optimization patch soon.

- 1. suppress the above allocations with UnsafeUtility
- 2. introduce a faster technique I found for iteration of subscribers.

Thanks for the your report!

**Comment by yenmoc** [comment]
Thank you, I forgot about the allocation the first time,

**Comment by hadashiA** [comment]
Merged #11.

Allocations were reduced to zero and made even faster.

<img width="787" alt="スクリーンショット 2024-01-13 12 05 23" src="https://github.com/hadashiA/VitalRouter/assets/727159/9084759b-5d38-4185-bd4b-bf7313b3b457">

Note that async builder allocations are reduced in Release builds.

<img width="782" alt="スクリーンショット_2024-01-13_12_09_59" src="https://github.com/hadashiA/VitalRouter/assets/727159/d4a2e6b6-0d6a-42d0-bc79-66dd0485f7ae">
